src/main/java/com/google/api/codegen/metacode/FieldStructureParser.java

```diff
@@ -50,26 +74,61 @@ static Pattern getFieldMapPattern() {
     return fieldMapPattern;
   }
 
-  public static InitCodeNode parse(String dottedPathString) {
-    return parse(dottedPathString, ImmutableMap.<String, InitValueConfig>of());
+  public static InitCodeNode parse(String initFieldConfigString) {
+    return parse(initFieldConfigString, ImmutableMap.<String, InitValueConfig>of());
   }
 
   public static InitCodeNode parse(
-      String dottedPathString, Map<String, InitValueConfig> initValueConfigMap) {
-    String[] equalsParts = dottedPathString.split("[=]");
+      String initFieldConfigString, Map<String, InitValueConfig> initValueConfigMap) {
+    InitFieldConfig fieldConfig = createInitFieldConfig(initFieldConfigString);
+    InitValueConfig valueConfig = null;
+
+    if (fieldConfig.hasInitValue()) {
+      valueConfig = createInitValueConfig(fieldConfig, initValueConfigMap);
+    } else if (initValueConfigMap.containsKey(fieldConfig.fieldPath())) {
+      valueConfig = initValueConfigMap.get(fieldConfig.fieldPath());
+    }
+
+    return parsePartialDottedPathToInitCodeNode(
+        fieldConfig.fieldPath(), InitCodeLineType.Unknown, valueConfig, null);
+  }
+
+  private static InitFieldConfig createInitFieldConfig(String initFieldConfigString) {
```

Comment:
I definitely agree about equalParts[], it is good to use the fieldName, entityName and value variables. But could they, and the splitting code, be part of the parse method, and be parsed as separate args to createInitValueConfig?